File: 2023/advent_of_code/test/q18/q18_test.exs
defmodule AdventOfCode.Q18Test do
  use ExUnit.Case
  doctest AdventOfCode.Q18

  test "Q18 Part 1 Example works" do
    file_stream = File.stream!("./test/q18/q18_example.txt")
    result = AdventOfCode.Q18.part1(file_stream)
    assert result == 62
  end

  test "Q18 Part 1 works" do
    file_stream = File.stream!("./test/q18/q18.txt")
    result = AdventOfCode.Q18.part1(file_stream)
    assert result == 26857
  end
end

File: 2022/AdventOfCode/src/test/kotlin/q18/Q18Test.kt
package q18

import junit.framework.TestCase.assertEquals
import java.nio.file.Paths
import kotlin.test.Test

class Q18Test {
    @Test
    fun `part 1 works`()  {
        val filePath = javaClass.getResource("/q18.txt")?.toURI()?.let { Paths.get(it) }
        assertEquals(3346, part1(filePath!!) )
    }

    @Test
    fun `part 2 works`()  {
        val filePath = javaClass.getResource("/q18.txt")?.toURI()?.let { Paths.get(it) }
        assertEquals(1980 , part2(filePath!!) )
    }
}